src/Model/PhasorDynamics/Bus/BusSignal.hpp

```diff
@@ -0,0 +1,127 @@
+
+#pragma once
+
+#include <Model/PhasorDynamics/BusBase.hpp>
+
+namespace GridKit
+{
+  namespace PhasorDynamics
+  {
+
+    // TODO change base class so there is no Vr and Vi
+
+    template <class ScalarT, typename IdxT>
+    class BusSignal : public BusBase<ScalarT, IdxT>
+    {
+      using BusBase<ScalarT, IdxT>::size_;
+      using BusBase<ScalarT, IdxT>::y_;
+      using BusBase<ScalarT, IdxT>::yp_;
+      using BusBase<ScalarT, IdxT>::yB_;
+      using BusBase<ScalarT, IdxT>::ypB_;
+      using BusBase<ScalarT, IdxT>::f_;
+      using BusBase<ScalarT, IdxT>::fB_;
+      using BusBase<ScalarT, IdxT>::tag_;
+
+    public:
+      using real_type = typename BusBase<ScalarT, IdxT>::real_type;
+      using DataT     = BusData<real_type, IdxT>;
+
+      BusSignal(const DataT& data);
+
+      ~BusSignal()
+      {
+      }
+
+      virtual int allocate() override;
+      virtual int tagDifferentiable() override;
+      virtual int initialize() override;
+      virtual int evaluateResidual() override;
+      virtual int evaluateJacobian() override;
+
+      /**
+       * @brief A one-time initialization function
+       * that can be called by any.
+       *
+       * @return state value of signal
+       */
+      void initial_value(ScalarT value) override
+      {
+
+        y_[0] = value;
+        // if(is_initialized_)
+        //{
+        //   std::cout << "ERROR!";
+        // }
+        // else{
+        //   y_[0] = value;
+        //   is_initialized_ = true;
+        // }
```

Comment:
why is this commented out? it appears to implement the intended behavior but instead we always set `y_[0]` to the value provided.